pkg/scheduler/partition.go

```diff
@@ -616,8 +613,6 @@ func (pc *PartitionContext) addNodeToList(node *objects.Node) error {
 // removeNodeFromList removes the node from the list of partition nodes.
 // This locks the partition.
 func (pc *PartitionContext) removeNodeFromList(nodeID string) *objects.Node {
```

Comment:
The function comment should be updated. The lock responsibility has been moved down to pc.nodes. No partition lock here.